### PR TITLE
For #1457: Implemented RtBranches.find and removed annotation from test

### PR DESCRIPTION
**Pull request by paulodamaso**
For #1457: Implemented RtBranches.find and removed annotation from test

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#1458](https://github.com/jcabi/jcabi-github/issues/1458) is now in scope, role is `REV`

File: src/main/java/com/jcabi/github/RtBranches.java [hunk 1]
```diff
@@ -103,6 +104,8 @@ public Branch map(final JsonObject object) {
 
     @Override
     public Branch find(final String name) {
-        throw new UnsupportedOperationException("find(name) not supported");
+        return StreamSupport.stream(this.iterate().spliterator(), false).filter(
+            item -> item.name().equals(name)
+        ).findFirst().orElse(null);
```
Comment (amihaiemil) [inline src/main/java/com/jcabi/github/RtBranches.java]: @paulodamaso Should we really return ``null`` if we don't find it? What does the Github API say, maybe we can create it if we don't find an existing one?

Comment (paulodamaso) [reply src/main/java/com/jcabi/github/RtBranches.java]: @amihaiemil Unfortunately all we have on Github API is https://developer.github.com/v3/repos/branches/#get-branch , and it doesn't say nothing about a seearch without result. I did not liked this `null` either, but I don't know if it is right to create a new branch right here on this method; maybe we could implement something like a `findOrCreate´ method later, WDYT?

**Comment by paulodamaso** [comment]
@amihaiemil Please take a look at the answer

**Comment by amihaiemil** [comment]
@rultor merge it

**Comment by 0crat** [comment]
Job `gh:jcabi/jcabi-github#1458` is not assigned, can't get performer

**Comment by 0crat** [comment]
The job [#1458](https://github.com/jcabi/jcabi-github/issues/1458) is now out of scope

**Pull request merged**